File: src/main/java/org/yinwang/pysonar/hash/FunTypeEqualFunction.java
package org.yinwang.pysonar.hash;

import org.yinwang.pysonar.types.FunType;


public class FunTypeEqualFunction extends EqualFunction {

	public boolean equals(Object x, Object y) {
		if (x instanceof FunType && y instanceof FunType) {
			FunType xx = (FunType) x;
			FunType yy = (FunType) y;
			return xx == yy ||
					xx.getTable().getPath().equals(yy.getTable().getPath());
		} else {
			return x.equals(y);
		}
	}
}

File: src/main/kotlin/org/yinwang/pysonar/hash/FunTypeEqualFunction.kt
package org.yinwang.pysonar.hash

import org.yinwang.pysonar.types.FunType

class FunTypeEqualFunction : EqualFunction {
	override fun equals(x: Any, y: Any): Boolean {
		return if (x is FunType && y is FunType) {
			x === y || x.table.path == y.table.path
		} else {
			x == y
		}
	}
}
